<robot name="test_robot">
  <link name="base_link">
        <visual>
                <geometry>
                        <cylinder length="0.6" radius="0.2"/>
                </geometry>
        </visual>
  </link>

  <link name="arm1">
	<visual>
		<geometry>
			<box size = "2 .1 .1"/>
		</geometry>
		<origin xyz = "-1 0 0"/>
	</visual>
  </link>
    
  <link name="link1">
	<visual>
		<geometry>
			<cylinder length="0.4" radius="0.1"/>
		</geometry>
	</visual>
  </link>

  <joint name="link1_to_arm1" type="fixed">
    <parent link = "link1"/>
    <child link = "arm1"/>
  </joint>

  <link name="arm2">
        <visual>
                <geometry>
                        <box size = "1 .1 .1"/>
                </geometry>
                <origin xyz = "-0.5 0 0"/>
        </visual>
  </link>


  <link name="link2">
        <visual>
                <geometry>
                        <cylinder length="0.4" radius="0.1"/>
                </geometry>
        </visual>
  </link>

  <joint name="link2_to_arm2" type="fixed">
    <parent link = "link2"/>
    <child link = "arm2"/>
  </joint>

  <link name="arm3">
        <visual>
                <geometry>
                        <box size = ".02 .02 3"/>
                </geometry>
                <origin xyz = "0 0 -1.5"/>
        </visual>
  </link>


  <link name="link3">
        <visual>
                <geometry>
                        <sphere radius="0.1"/>
                </geometry>
        </visual>
  </link>

  <joint name="link3_to_arm3" type="fixed">
    <parent link = "link3"/>
    <child link = "arm3"/>
  </joint>


  <joint name="joint1" type="fixed">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="1 1.73025 0" rpy="0 -0 1.0472" />
  </joint>

  <joint name="joint2" type="fixed">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0.92388 0.382683 0" rpy="3.14159 -0 0.392699" />
  </joint>

  <joint name="joint3" type="fixed">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0 0 2" rpy="0 0 0" />
  </joint>
</robot>
